File: java-backend/src/main/java/com/michael21/SoundFilter/audio/data/TranscriptionResult.java
package com.michael21.SoundFilter.audio.data;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import java.util.List;

@Data
public class TranscriptionResult {
    private String filename;
    private String transcript;
    private List<WordTimestamp> words;

    @JsonProperty("detected_language")
    private String detectedLanguage;

    @JsonProperty("processing_time")
    private double processingTime;

    @Data
    public static class WordTimestamp {
        private String word;

        @JsonProperty("start_time")
        private double startTime;

        @JsonProperty("end_time")
        private double endTime;
    }
}

File: frontend/models/audio/TranscriptionResult.ts
export interface TranscriptionResult {
    filename: string
    transcript: string
    words: WordTimestamp[]
    detectedLanguage: string | null
    processingTime: number
  }
  
  export interface WordTimestamp {
    word: string
    // Support multiple possible property names for timestamps
    startTime?: number | string
    endTime?: number | string
    start_time?: number | string // Snake case alternative
    end_time?: number | string // Snake case alternative
  }
  
  